code/game/objects/structures/watercloset.dm

```diff
@@ -486,8 +488,26 @@
 		new_sink.setDir(dir)
 		qdel(src)
 		return
+	if(I.tool_behaviour == TOOL_WRENCH && !(flags_1&NODECONSTRUCT_1))
+		I.play_tool_sound(src)
+		deconstruct()
+		return
+
 	return ..()
 
+/obj/structure/sinkframe/deconstruct()
+	if(!(flags_1 & NODECONSTRUCT_1))
+		drop_materials()
+	..()
+
+/obj/structure/sinkframe/proc/drop_materials()
+	if(buildstacktype)
+		new buildstacktype(loc,buildstackamount)
+	else
+		for(var/i in custom_materials)
+			var/datum/material/M = i
```

Comment:
You should use as anything here, see <https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#avoid-unnecessary-type-checks-and-obscuring-nulls-in-lists>
```suggestion
		for(var/datum/material/mat in custom_materials)
```
Oh also you should use a variable name longer then a single letter, it makes things easier to parse, especially if that name has some meaning that better explains what  something does